tsdb/chunkenc/histogram.go

```diff
@@ -288,34 +289,35 @@ func (a *HistogramAppender) appendable(h *histogram.Histogram) (
 
 	if h.Count < a.cnt {
 		// There has been a counter reset.
-		counterReset = true
+		counterResetHeader = CounterReset
 		return
 	}
 
 	if h.Schema != a.schema || h.ZeroThreshold != a.zThreshold {
+		counterResetHeader = UnknownCounterReset
```

Comment:
Let's add a comment that this case might or might not go along with a counter reset and that we do not want to invest the work of a full counter reset detection as long as #15346 is still open. And maybe even add a TODO to consider adding the counter reset detection here once #15346 is fixed.

Reply:
Done